commons-rng-core/src/main/java/org/apache/commons/rng/core/source64/SplitMix64.java

```diff
@@ -59,7 +59,7 @@ public SplitMix64(Long seed) {
      * @param seed Seed.
      */
     private void setSeedInternal(Long seed) {
-        state = seed.longValue();
+        this.state = seed;
```

Comment:
This explicit use of `longValue` was discussed when the constructor was added to use a primitive long. The code as is makes it explicit that the input is a Long and not a long. Looking at this now the entire setSeedInternal method is redundant. It is preserved to match other implementations. There are 9 classes in the package with the setSeedInternal method that use the constructor input object argument named seed. The method will be inlined by the JVM anyway (it is less than 35 bytes when in java byte code) and the performance impact is negligible.